python_agent/custom_env.py

```
@@ -131,107 +157,249 @@ def _get_info(self):
+        
+        player_x, player_y, player_plane = player_loc_data[0], player_loc_data[1], player_loc_data[2]
+
+        for i in range(MAX_NEARBY_NPCS):
+            npc_id = nearby_npcs[i, 0]
+            npc_x = nearby_npcs[i, 1]
+            npc_y = nearby_npcs[i, 2]
+            # npc_anim = nearby_npcs[i, 3] # NPC animation (could be used to check if already dead/fighting)
+
+            if npc_id == GOBLIN_NPC_ID:
+                # Basic check: is NPC data valid (not -1 padding)?
+                if npc_x != -1 and npc_y != -1: # Assuming NPCs are on the same plane as player
```

Ruff (0.11.9)

Use a single `if` statement instead of nested `if` statements

(SIM102)